components/mpas-ocean/src/Registry.xml

```diff
@@ -3760,6 +3807,10 @@
 			 description="Fresh water flux from river runoff at cell centers from coupler. Positive into the ocean."
 			 packages="thicknessBulkPKG"
 		/>
+		<var name="subglacialRunoffFlux" type="real" dimensions="nCells Time" units="m^3 s^-1"
```

Comment:
I would really rather have this be the same units, i.e., that all of these be surface mass fluxes rather than one being a surface volume flux.

Reply:
You mean riverRunoffFlux have same units as subglacialRunoffFlux? Yes, I thought about that too, but I think this depends on the model that provides these inputs and the units it does that in. The subglacialRunoffFlux is provide by MALI which gives me volume flux. To get that into the same units as riverRunoffFlux, I would have to multiply by density during preprocessing and divide by density again in mpas_ocn_surface_bulk_forcing.F, which leaves some room for error (e.g. using slightly different values during preprocessing and then in the mpas code). If you think it is too confusing as is, I could rename subglacialRunoffFlux to subglacialRunoffVolumeFlux if that would help?

Reply:
The file Alex Hager (can't tag him for some reason) gave me has totalGroundingLineDischargeCell in m^3/s maybe him or @matthewhoffman can comment on this?

Reply:
The density values are not supposed to be consistent. Mass fluxes into MPAS-Ocean are assumed to have the reference density, and that's what we should be dividing by, while on the MALI side we should be multiplying by the density of freshwater, as Xylar states.

Reply:
Yes, exactly.

Reply:
MALI calculates that field by summing the volumes fluxes from both the distributed and channelized systems. So I just used the units hardcoded into MALI without multiplying by density. MALI uses a freshwater density of 1000 kg/m3 though, so you can use that density to convert to a mass flux

Reply:
Ok, I will just divide by freshwater density and areaCell during preprocessing so that in the MPAS-O code subglacialRunoffFlux has same units and is treated exactly the same way as riverRunoffFlux